pkg/controllers/cspc-controller/sync.go

```diff
@@ -261,3 +265,60 @@ func (c *Controller) GetCSPIWithoutDeployment(cspc *cstor.CStorPoolCluster) ([]c
 	}
 	return cspiList, nil
 }
+
+// syncCSPI propagates all the required changes from cspc to respective cspi.
+func (pc *PoolConfig) syncCSPI(cspc *cstor.CStorPoolCluster) {
+	cspiList, err := pc.Controller.GetCSPIListForCSPC(cspc)
+	if err != nil {
+		klog.Errorf("failed to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+	if len(cspiList.Items) == 0 {
+		klog.Errorf("No cspi(s) found while trying to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+
+	for _, cspi := range cspiList.Items {
+		cspi := cspi
+		err := pc.syncCSPIWithCSPC(cspc, &cspi)
+		if err != nil {
+			klog.Errorf("failed to sync cspi %s from its parent cspc %s", cspi.Name, cspc.Name)
+		}
+	}
+}
+
+func (pc *PoolConfig) syncCSPIWithCSPC(cspc *cstor.CStorPoolCluster, cspi *cstor.CStorPoolInstance) error {
+	cspiCopy := cspi.DeepCopy()
+	klog.V(2).Infof("Syncing cspi %s from parent cspc %s", cspiCopy.Name, cspc.Name)
+	for _, poolSpec := range cspc.Spec.Pools {
+		poolSpec := poolSpec
+		cspiCopy.Spec.PoolConfig = poolSpec.PoolConfig
+		defaultPoolConfig(cspiCopy, cspc)
+	}
+
+	if reflect.DeepEqual(cspiCopy, cspi) {
+		gotCSPI, err := pc.Controller.GetStoredCStorVersionClient().CStorPoolInstances(cspiCopy.Namespace).Update(cspiCopy)
+		if err != nil {
+			return errors.Errorf("Failed to sync cspi %s from parent cspc %s", cspiCopy.Name, cspc.Name)
+		}
+		_, err = pc.Controller.kubeclientset.AppsV1().Deployments(gotCSPI.Namespace).Update(pc.AlgorithmConfig.GetPoolDeploySpec(gotCSPI))
```

Comment:
Is it good to do patch instead of updating deployment(It will lose user-configured labels or annotations -- before that is it Valid use case)?

Reply:
But do we want to make the pool-manager deployment to be user tunable separately? 
In fact, I thought about this and choose to go with `update` instead of `patch` intentionally.
Thinking on lines of gitops. 
Let us see what @kmova and @vishnuitta have to say about this. 
If we need to make sure that user configured labels/annotations are not lost -- I can do it a strategic merge patch.

IMO -- the pool-manger config should be only dependent on cspi-config and must not be tuned externally but your point of not losing user set annotations and labels also look reasonable. Can discuss and finalize this.